### PR TITLE
Enable linking in exception throwing support library when build onnxruntime wasm.

### PR DESCRIPTION
**Pull request by Zuwei-Zhao**
**Description**: Enable linking in exception throwing support library when build onnxruntime wasm containing extensions using exceptions.
**Motivation and Context**
- Currently when we build onnxruntime **wasm** and **disable exceptions**, it will also **disable linking in exception throwing support library** in emscripten. The issue is also described [here](https://github.com/emscripten-core/emscripten/issues/10060);
- Because onnxruntime-extensions now has some submodules which are still **using exceptions**, so the linking will fail;
- Add a build option: `--enable_wasm_exception_throwing_override` and reset the link flag to enable linking exception support library. The description of this flag is here: https://github.com/emscripten-core/emscripten/blob/ad6e356df588646eb2569c8e9100b8991db799e8/src/settings.js#L1819.

### CODE REVIEW

File: cmake/onnxruntime_webassembly.cmake [hunk 1]
```diff
@@ -61,6 +61,12 @@ else()
   set_property(TARGET onnxruntime_webassembly APPEND_STRING PROPERTY LINK_FLAGS " -s ASSERTIONS=0 -s SAFE_HEAP=0 -s STACK_OVERFLOW_CHECK=0 -s DEMANGLE_SUPPORT=0")
 endif()
 
+# Set link flag for onnxruntime-extensions because now it contains submodules using exceptions
+# Would remove this link flag when all submodules have supported disabling c++ exceptions
+if (onnxruntime_USE_EXTENSIONS)
```
Comment (wenbingl) [inline cmake/onnxruntime_webassembly.cmake:66]: And disable-exception flag?

Comment (Zuwei-Zhao) [reply cmake/onnxruntime_webassembly.cmake]: Hi @wenbingl and @fs-eire , according to your advice, I have added a flag to build and enable linking exceptions library when build with `--disable_exceptions`. Please help review, thanks.

**Review by wenbingl — DISMISSED** [review]

**Comment by wenbingl** [comment]
@fs-eire , can you take a look as well?

File: tools/ci_build/build.py [hunk 2]
```diff
@@ -756,7 +760,8 @@ def generate_build_tree(cmake_path, source_dir, build_dir, cuda_home, cudnn_home
         "-Donnxruntime_MINIMAL_BUILD=" + ("ON" if args.minimal_build is not None else "OFF"),
         "-Donnxruntime_EXTENDED_MINIMAL_BUILD=" + ("ON" if args.minimal_build and 'extended' in args.minimal_build
                                                    else "OFF"),
-        "-Donnxruntime_MINIMAL_BUILD_CUSTOM_OPS=" + ("ON" if args.minimal_build and 'custom_ops' in args.minimal_build
+        "-Donnxruntime_MINIMAL_BUILD_CUSTOM_OPS=" + ("ON" if (args.minimal_build is not None and ('custom_ops' in
+                                                     args.minimal_build or args.use_extensions))
```
Comment (Zuwei-Zhao) [inline tools/ci_build/build.py:764]: Hi @wenbingl , I updated the argument here. Please help review. Thanks.

File: tools/ci_build/build.py [hunk 3]
```diff
@@ -794,6 +799,8 @@ def generate_build_tree(cmake_path, source_dir, build_dir, cuda_home, cudnn_home
         "-Donnxruntime_ENABLE_WEBASSEMBLY_SIMD=" + ("ON" if args.enable_wasm_simd else "OFF"),
         "-Donnxruntime_ENABLE_WEBASSEMBLY_EXCEPTION_CATCHING=" + ("OFF" if args.disable_wasm_exception_catching
                                                                   else "ON"),
+        "-Donnxruntime_ENABLE_WEBASSEMBLY_EXCEPTION_THROWING=" + ("ON" if args.enable_wasm_exception_throwing_override
```
Comment (Zuwei-Zhao) [inline tools/ci_build/build.py:802]: Hi @fs-eire , please be aware of the change here concerning wasm build. Thanks.

**Review by fs-eire — DISMISSED** [review]
Looks good to me for WebAssembly part.

**Review by fs-eire — APPROVED** [review]

**Review by wenbingl — APPROVED** [review]

**Comment by Zuwei-Zhao** [comment]
"iOS CI Pipeline" failed 3 times due to time limits, ignore it now because it's optional.

**Pull request merged**